File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/interpreter/lab/BinaryMinus.m
% BinaryMinus.m

classdef BinaryMinus < Binary
	methods
		% constructor:
		function theBinaryMinus = BinaryMinus(aOperand1, aOperand2)
			theBinaryMinus@Binary(aOperand1, aOperand2);
		end

		function output(theBinaryMinus)
			theBinaryMinus.outputInfix('-');
		end
	end
end

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/lab/BinaryMinus.java
// BinaryMinus.java

import java.io.PrintWriter;

public class BinaryMinus extends Binary
{
	public BinaryMinus(Expression aOperand1, Expression aOperand2)
	{
		super(aOperand1, aOperand2);
	}

	public void output(PrintWriter aPrintWriter)
	{
		outputInfix(aPrintWriter, "-");
	}
}
